typings/internal.d.ts

```diff
@@ -60,13 +60,19 @@ interface FrontendSettings {
 }
 
 interface AppSettings {
+    applicationFolder:string,
+    settingsFolder:string,
+    settingsFile:string,
+    attachedExtensionsFile:string,
+    frontendSettings:FrontendSettings,
     getApplicationFolder():string,
-    validate():AppSettings,
-    getId():string,
-    setId(id:string):AppSettings,
-    attachExtension(pathName:string,extensionInfo:ExtensionInfo,force?:boolean):AppSettings,
-    detachExtension(extensionId:string):AppSettings,
-    detachAllExtensions():AppSettings,
+    validate():Promimse<AppSettings>,
+    getId():Promimse<string>,
+    setId(id:string):Promise<AppSettings>,
+    attachExtension(pathName:string,extensionInfo:ExtensionInfo,force?:boolean):Promise<AppSettings>,
+    detachExtension(extensionId:string):Promise<AppSettings>,
+    detachAllExtensions():Promise<AppSettings>,
+    loadAttachedExtensions():Promise<Object.<string, AttachedExtension>>,
```

Comment:
The notation with a . apparently is not valid. This seems to be valid but I'm unsure if expressing the right thing:
Promise<string, AttachedExtension>